conans/client/installer.py

```diff
@@ -454,3 +456,15 @@ def _call_package_info(self, conanfile, package_folder, is_editable):
                 self._hook_manager.execute("post_package_info", conanfile=conanfile)
 
         conanfile.cpp_info.check_component_requires(conanfile)
+
+    def _call_install_method(self, conanfile, install_folder):
+        if hasattr(conanfile, "install"):
+            conanfile.folders.set_install_folder(install_folder)
```

Comment:
This makes it so that `self.install_folder` is available inside the `install()` method (and beyond, but there's not much use for it after the method has been called) This will also set the `self.immutable_package_folder` to the original `package_folder` so that the information is not lost